src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -1346,6 +1346,147 @@ static void G(IOut<(string, string)> x, IOut<(string, string?)> y, IOut<(string?
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "F(w)").WithLocation(28, 9));
         }
 
+        [Fact]
+        public void TupleTypeInference_06()
+        {
+            var source =
+@"class C
+{
+    static void F(object? x, object? y)
+    {
+        if (y != null)
+        {
+            (object?, object?) t = (x, y);
+            t.Item1.ToString();
+            t.Item2.ToString();
+        }
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            // PROTOTYPE(NullableReferenceTypes): Should not report warning for `t.Item2`.
+            comp.VerifyDiagnostics(
+                // (8,13): warning CS8602: Possible dereference of a null reference.
+                //             t.Item1.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Item1").WithLocation(8, 13),
+                // (9,13): warning CS8602: Possible dereference of a null reference.
+                //             t.Item2.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Item2").WithLocation(9, 13));
+
+        }
```

Comment:
Nit: empty line above.
Not necessarily for this PR, but it's good to test nested tuples, long tuples, and named tuple elements.